server/server_test.go

```diff
@@ -339,11 +340,112 @@ func runTestPreparedTimestamp(t *C) {
+
+	runTestsOnNewDB(c, func(config *mysql.Config) {
+		config.AllowAllFiles = true
+		config.Strict = false
+	}, "LoadData2", func(dbt *DBTest) {
+		dbt.mustExec("create table t (a int, b int default -1, c binary(1), id int not null auto_increment, primary key(id))")
+		rs, err := dbt.db.Exec("load data local infile '/tmp/load_data_test2.csv' into table t fields terminated by ' ' (@x, b, @dummy, @z) set a = @x, c = unhex(@z)")
+		dbt.Assert(err, IsNil)
+		affectedRows, err := rs.RowsAffected()
+		dbt.Assert(err, IsNil)
+		dbt.Assert(affectedRows, Equals, int64(4))
+
+		var (
+			aa *int
+			bb *int
+			cc []uint8
+		)
+		rows := dbt.mustQuery("select a, b, c from t")
+		dbt.Check(rows.Next(), IsTrue, Commentf("unexpected data"))
+		err = rows.Scan(&aa, &bb, &cc)
+		dbt.Check(err, IsNil)
+		dbt.Check(*aa, DeepEquals, 0)
+		dbt.Check(bb, DeepEquals, (*int)(nil))
+		dbt.Check(cc, DeepEquals, ([]uint8)(nil))
+		dbt.Check(rows.Next(), IsTrue, Commentf("unexpected data"))
+		err = rows.Scan(&aa, &bb, &cc)
+		dbt.Check(err, IsNil)
+		dbt.Check(*aa, DeepEquals, 11)
+		dbt.Check(*bb, DeepEquals, 12)
+		dbt.Check(cc, DeepEquals, []byte{1})
+		dbt.Check(rows.Next(), IsTrue, Commentf("unexpected data"))
+		err = rows.Scan(&aa, &bb, &cc)
+		dbt.Check(err, IsNil)
+		dbt.Check(rows.Next(), IsTrue, Commentf("unexpected data"))
+		err = rows.Scan(&aa, &bb, &cc)
+		dbt.Check(err, IsNil)
+		dbt.Check(rows.Next(), IsFalse, Commentf("unexpected data"))
+
+		dbt.mustExec("truncate t")
+
+		rs, err = dbt.db.Exec("load data local infile '/tmp/load_data_test2.csv' into table t fields terminated by ' ' (a) set a = a, b = a, c = unhex(1)")
+		dbt.Assert(err, IsNil)
+		affectedRows, err = rs.RowsAffected()
+		dbt.Assert(err, IsNil)
+		dbt.Assert(affectedRows, Equals, int64(4))
+		rows = dbt.mustQuery("select a, b, c from t")
+		dbt.Check(rows.Next(), IsTrue, Commentf("unexpected data"))
+		err = rows.Scan(&aa, &bb, &cc)
+		dbt.Check(err, IsNil)
+		dbt.Check(*aa, DeepEquals, 0)
+		dbt.Check(*bb, DeepEquals, 0)
+		//dbt.Check(cc, DeepEquals, ([]uint8)(nil)) FIXME: stranger
+		dbt.Check(rows.Next(), IsTrue, Commentf("unexpected data"))
+		err = rows.Scan(&aa, &bb, &cc)
+		dbt.Check(err, IsNil)
+		dbt.Check(*aa, DeepEquals, 11)
+		dbt.Check(*bb, DeepEquals, 11)
+		dbt.Check(cc, DeepEquals, []byte{1})
+		dbt.Check(rows.Next(), IsTrue, Commentf("unexpected data"))
+		err = rows.Scan(&aa, &bb, &cc)
+		dbt.Check(err, IsNil)
+		dbt.Check(*aa, DeepEquals, 21)
+		dbt.Check(*bb, DeepEquals, 21)
+		dbt.Check(cc, DeepEquals, []byte{1})
+		dbt.Check(rows.Next(), IsTrue, Commentf("unexpected data"))
+		dbt.Check(rows.Next(), IsFalse, Commentf("unexpected data"))
+
+		startTime := time.Now()
+		time.Sleep(1 * time.Second)
+		dbt.mustExec("drop table if exists t2")
+		dbt.mustExec("create table t2 (a int, b timestamp)")
+		rs, err = dbt.db.Exec("load data local infile '/tmp/load_data_test2.csv' into table t2 fields terminated by ' ' set a = (select 10), b = current_timestamp")
```

Comment:
plz add a test like `set a = default`

Reply:
we don't support `default`, not only `load data set` but also `insert on duplicate key`  need support `default` expression, maybe we can do it in another issue/PR after this one merged? :laughing:

Reply:
OK